src/comps/Product.jsx

```diff
@@ -1,36 +1,66 @@
-import React, { useContext, useState } from "react";
+import React, { useContext } from "react";
 import productCss from "../styles/product.module.css";
 import axios from "axios";
 import { contextApi } from "../contextApi";
 export default function Prodact(props) {
   const valContext = useContext(contextApi);
-  const [refreshPage,setRefreshPage]=useState(false)
   const addToCart = async () => {
-    
     const res = await axios.post("http://localhost:3001/addToCart", {
       productId: props.val._id,
-      userNameId: valContext._id,
+      userNameId: valContext.userData._id,
     });
     const data = await res.data;
     if (data === "product added to cart!") {
-      setRefreshPage(!refreshPage)
       alert(data);
+      valContext.addToCart(props.val);
     } else {
       console.log("cannot added to cart");
     }
   };
+  const deleteIteam = async () => {
+    try {
+      const res = await axios.post("http://localhost:3001/deleteIteam", {
+        productId: props.val._id,
+        userNameId: valContext.userData._id,
+      });
+      const data = await res.data;
+      if (data === "Product deleted from cart") {
+        alert(data);
+        valContext.deleteFromCart(props.index)
+      } else {
+        console.error(data);
+      }
+    } catch (err) {
+      console.log(err);
+    }
+    props.refresh();
+  };
   return (
     <div className={productCss.singleProduct}>
-      <h2>{props.val.name}</h2>
-      <h2>{props.val.description}</h2>
-      <h2>{props.val.price}</h2>
-      <img src={props.val.image} className={productCss.img} alt={props.val.name} />
-      {props.url === "getAddToCart" && (
-        <h1 className={productCss.plus} onClick={addToCart}>
-          
+      {props.url === "cart" && (
+        <h1
+          className={productCss.plusOrX}
+          id={productCss.delete}
+          onClick={deleteIteam}
```

Comment:
Can you think of a better name for this class name? delete is very generic, not telling the reader any thing about the meaning or purpose of the element with this id